static/app/utils/replays/playback/hooks/useReplayCurrentTime.tsx

```diff
@@ -0,0 +1,37 @@
+import {useEffect} from 'react';
+
+import useReplayPlayerState from 'sentry/utils/replays/playback/providers/useReplayPlayerState';
+import useRAF from 'sentry/utils/useRAF';
+
+interface Props {
+  callback: (props: {timeMs: number}) => void;
+}
+
+export default function useReplayCurrentTime({callback}: Props) {
+  const {isFinished, playerState, replayers} = useReplayPlayerState();
+  const replayer = replayers.at(0);
+  const state = replayer?.service.state;
+
+  useRAF(
+    () => {
+      if (replayer) {
+        callback({timeMs: replayer.getCurrentTime()});
+      }
+    },
+    {enabled: Boolean(replayer) && playerState === 'playing'}
+  );
+
+  useEffect(() => {
+    if (state?.value === 'paused' && state?.context.timeOffset !== undefined) {
+      if (isFinished && replayer) {
+        callback({timeMs: replayer?.getCurrentTime()});
+      } else {
+        callback({timeMs: state?.context.timeOffset});
+      }
```

Comment:
lgoaf low, but this helps to see what is conditional

```suggestion
      callback({
        timeMs: isFinished && replayer
          ? replayer?.getCurrentTime()
          : state?.context.timeOffset
      });
```